### PR TITLE
NAS-107351 / 12.1 / datasets *or zvols* in several places, and a little comment on overlapping tasks

### PR DESCRIPTION
**Pull request by stilez**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/snapshot.py [hunk 1]
```diff
@@ -95,11 +95,13 @@ async def do_create(self, data):
         """
         Create a Periodic Snapshot Task
 
-        Create a Periodic Snapshot Task that will take snapshots of specified `dataset` at specified `schedule`.
+        Create a Periodic Snapshot Task that will take snapshots of specified `dataset` or `zvol` at specified `schedule`.
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/snapshot.py:98]: This is a parameter name, should remain unchanged.

For ZFS perspective everything is a dataset, what we call dataset is a dataset of FILESYSTEM type, zvol is a dataset of VOLUME type.

Comment (stilez) [reply src/middlewared/middlewared/plugins/snapshot.py:98]: The target audience isn't people who think technically about ZFS and its developer terminology. It's people who use the web UI, and in the web UI we universally do distinguish datasets and volumes that way. So stating only datasets here strongly implies "not volumes" to an end user.

Comment (william-gr) [reply src/middlewared/middlewared/plugins/snapshot.py:98]: This is the API, text not available to end users. There is no `zvol` attribute in the API.

Comment (stilez) [reply src/middlewared/middlewared/plugins/snapshot.py:98]: Damn. I wish I understood the codebase better :) Id like to offer direct PRs not just ask others to do the work. would that be okay?

Comment (william-gr) [reply src/middlewared/middlewared/plugins/snapshot.py:98]: Any contribution in actual code will always be very welcome ;)
As long as you're comfortable receiving harsh feedback of people with OCD :D

Comment (stilez) [reply src/middlewared/middlewared/plugins/snapshot.py:98]: And as long as you're okay reviewing rubbish PRs from people who don't really have a clue! :)

Comment (william-gr) [reply src/middlewared/middlewared/plugins/snapshot.py:98]: I can approve as soon as you revert that line ; )

File: src/middlewared/middlewared/plugins/snapshot.py [hunk 2]
```diff
@@ -95,11 +95,13 @@ async def do_create(self, data):
...
         Recursive snapshots can be created if `recursive` flag is enabled. You can `exclude` specific child datasets
-        from snapshot.
+        or zvols from the snapshot.
         Snapshots will be automatically destroyed after a certain amount of time, specified by
         `lifetime_value` and `lifetime_unit`.
+        If multiple periodic tasks create snapshots at the same time (for example hourly and daily at 00:00) the snapshot
+        will be kept until the last of these tasks reaches its expiry time.
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/snapshot.py:104]: There is a whitespace at the end of this line too.

File: src/middlewared/middlewared/plugins/snapshot.py (hunk 1 above)
Comment (stilez) [reply src/middlewared/middlewared/plugins/snapshot.py]: Reverted. Don't know github well enough to see it, but it' not letting me re-revert it so hopefully it agrees its done. If not please do fix.

File: src/middlewared/middlewared/plugins/snapshot.py (hunk 2 above)
Comment (stilez) [reply src/middlewared/middlewared/plugins/snapshot.py]: Space fixed. As above, can't see but believe its done.

**Comment by william-gr** [comment]
Thanks!

**Pull request merged**